src/mod/say/mod_say_nb/mod_say_nb.c

```diff
@@ -0,0 +1,676 @@
+ * Jakob Sundberg <jakob@mress.se> (2013)
+ *
+ * This is based on mod_say_sv.c
+ * mod_say_nb.c -- Say for Norwegian (Bokmål)
+ *
+ */
+
+#include <switch.h>
+#include <math.h>
+#include <ctype.h>
+
+SWITCH_MODULE_LOAD_FUNCTION(mod_say_nb_load);
+SWITCH_MODULE_DEFINITION(mod_say_nb, mod_say_nb_load, NULL, NULL);
+
+#define say_num(_sh, num, meth) {									\
+		char tmp[80];									\
+		switch_status_t tstatus;							\
+		switch_say_method_t smeth = say_args->method;					\
+		switch_say_type_t stype = say_args->type;					\
+		say_args->type = SST_ITEMS; say_args->method = meth;				\
+		switch_snprintf(tmp, sizeof(tmp), "%u", (unsigned)num);				\
+		if ((tstatus =									\
+			nb_say_general_count(_sh, tmp, say_args))				\
+			!= SWITCH_STATUS_SUCCESS) {						\
+			return tstatus;								\
+		}										\
+		say_args->method = smeth; say_args->type = stype;				\
+	}											\
+
+static switch_status_t play_group(switch_say_method_t method, switch_say_gender_t gender, int a, int b, int c, char *what, switch_say_file_handle_t *sh)
+{
+	/*
+	 * Norweigian, like swedish, have gendered numbers for the number one. Utrum is 'en' while neutrum is 'ett'. They also have 'et' but we'll probably
+	 * never encounter that, that's for when describing things like 'a house' - 'et hus' in contrast to specifically one house 'ett hus'
+	 * 
+	 * Source: https://www.riksmalsforbundet.no/grammatikk/kapittel-7-tallord/ 
+	 */
+	if (a) {
+		if (method == SSM_COUNTED) {
+			if ( a > 1 && b == 0 && c == 0) {	/* [2-9]00 */
+				switch_say_file(sh, "digits/%d", a); //to till ni
+			}
+			switch_say_file(sh, "digits/r-100"); //hundrede TODO: missing if we're ever going to need it
+		} else {
+			if (a == 1) {	/* 1xx */
+				switch_say_file(sh, "digits/n-1"); //ett
+			} else {	/* [2-9]xx */
+				switch_say_file(sh, "digits/%d", a);
+			}
+			switch_say_file(sh, "digits/100"); //hundre
+		}
+	}
+
+	if (b) {
+		if (b > 1) {	/* 20 <= 99 */
+			if (c == 0 && method == SSM_COUNTED) {
+				switch_say_file(sh, "digits/r-%d0", b);
+			} else {
+				switch_say_file(sh, "digits/%d0", b);
+			}
+		} else {	/* 10 < 20 */
+			if (method == SSM_COUNTED) {
+				switch_say_file(sh, "digits/r-%d%d", b, c);
+			} else {
+				switch_say_file(sh, "digits/%d%d", b, c);
+			}
+			c = 0; // reset c as we've already said this
+		}
+	}
+
+        if (c) {	/* 0 < 9 */
+		if (c == 1) {
+			if (what) {
+	 			switch_say_file(sh, "digits/n-1"); //ett
+			} else {
+				if (method == SSM_COUNTED) {
+					switch_say_file(sh, "digits/r-1"); //forsta
+				} else {
+					if (gender == SSG_UTRUM) {
+						switch_say_file(sh, "digits/u-1"); //ett
+					} else {
+						switch_say_file(sh, "digits/n-1"); //en
+					}
+				}
+			}
+		} else {
+			if (what) {
+				switch_say_file(sh, "digits/%d", c);
+			} else {
+				if (method == SSM_COUNTED) {
+					switch_say_file(sh, "digits/r-%d", c);
+				} else {
+					switch_say_file(sh, "digits/%d", c);
+				}
+			}
+		}
+	}
+
+        if (what && (a || b || c)) {
```

Comment:
don't mix indent. Goes for all files in this PR

Reply:
https://www.youtube.com/watch?v=2g5Hz17C4is